File: mobile/lib/data/services/prescription_service.dart
import 'package:dio/dio.dart';
import '../../core/constants/api_endpoints.dart';
import '../models/prescription_model.dart';

class PrescriptionService {
  PrescriptionService(this._dio);
  final Dio _dio;

  Future<PrescriptionAnalysisResultModel> upload(String filePath) async {
    final formData = FormData.fromMap({
      'file': await MultipartFile.fromFile(filePath, filename: filePath.split('/').last),
    });
    final response = await _dio.post(ApiPaths.prescriptionUpload, data: formData);
    return PrescriptionAnalysisResultModel.fromJson(response.data as Map<String, dynamic>);
  }

  Future<List<PrescriptionModel>> mine() async {
    final response = await _dio.get(ApiPaths.prescriptionsMine);
    return (response.data as List<dynamic>).map((e) => PrescriptionModel.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<List<PrescriptionModel>> queue() async {
    final response = await _dio.get(ApiPaths.prescriptionsQueue);
    return (response.data as List<dynamic>).map((e) => PrescriptionModel.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<PrescriptionModel> getById(int id) async {
    final response = await _dio.get(ApiPaths.prescriptionDetail(id));
    return PrescriptionModel.fromJson(response.data as Map<String, dynamic>);
  }

  Future<PrescriptionModel> updateItem(int itemId, Map<String, dynamic> payload) async {
    final response = await _dio.patch(ApiPaths.prescriptionItemUpdate(itemId), data: payload);
    return PrescriptionModel.fromJson(response.data as Map<String, dynamic>);
  }

  Future<PrescriptionModel> review(int id, {required String status, String? notes}) async {
    final response = await _dio.patch(ApiPaths.prescriptionReview(id), data: {
      'status': status,
      if (notes != null && notes.isNotEmpty) 'pharmacist_notes': notes,
    });
    return PrescriptionModel.fromJson(response.data as Map<String, dynamic>);
  }
}

File: frontend/services/prescription.service.ts
import { apiClient } from "@/lib/axios";
import type {
  Prescription,
  PrescriptionAnalysisResult,
  PrescriptionItemUpdatePayload,
  PrescriptionReviewPayload,
} from "@/types";

export const prescriptionService = {
  upload: async (file: File): Promise<PrescriptionAnalysisResult> => {
    const form = new FormData();
    form.append("file", file);
    const { data } = await apiClient.post<PrescriptionAnalysisResult>("/api/prescriptions/upload", form, {
      headers: { "Content-Type": "multipart/form-data" },
    });
    return data;
  },

  mine: async (): Promise<Prescription[]> => {
    const { data } = await apiClient.get<Prescription[]>("/api/prescriptions/mine");
    return data;
  },

  queue: async (): Promise<Prescription[]> => {
    const { data } = await apiClient.get<Prescription[]>("/api/prescriptions/queue");
    return data;
  },

  getById: async (id: number): Promise<Prescription> => {
    const { data } = await apiClient.get<Prescription>(`/api/prescriptions/${id}`);
    return data;
  },

  updateItem: async (itemId: number, payload: PrescriptionItemUpdatePayload): Promise<Prescription> => {
    const { data } = await apiClient.patch<Prescription>(`/api/prescriptions/items/${itemId}`, payload);
    return data;
  },

  review: async (id: number, payload: PrescriptionReviewPayload): Promise<Prescription> => {
    const { data } = await apiClient.patch<Prescription>(`/api/prescriptions/${id}/review`, payload);
    return data;
  },
};
